src/harness/testcases/heartbeat_testcase.py

```diff
@@ -824,3 +824,127 @@ def test_WINNF_FT_S_HBT_18(self):
                            grant_expire_times[response_num])
     self.assertEqual(response[2]['response']['responseCode'], 103)
     # No need to check transmitExpireTime because this is the first heartbeat
+
+    @winnforum_testcase
+    def test_WINNF_FT_S_HBT_19(self):
+        """Heartbeat Request from CBSD in Granted or Authorized state 
+        (immediately after CBSD moves into Granted State or following
+         a Heartbeat Response) requires CBSD to de-register. 
+
+        The response should be FAIL, code 105.
+        """
+
+        # Register the device
+        device_a = json.load(
+            open(os.path.join('testcases', 'testdata', 'device_a.json')))
+        self._sas_admin.InjectFccId({'fccId': device_a['fccId']})
+
+        request = {'registrationRequest': [device_a]}
+        response = self._sas.Registration(request)['registrationResponse']
```

Comment:
I don't think this and the next few lines work. Typically we will put "[0]" at the end of this line to grab the first response since there's only one. Then on line 845, there's no need to loop.